src/food/food.controller.ts

```diff
@@ -24,102 +24,111 @@ export class FoodController {
+      const food = this.foodService.setFoodCategory(
+        name,
+        subName,
+        level,
+        category,
+        foodId,
+      );
+
+      return food;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Get('/tests/result')
-  @ApiOperation({
-    summary: '사용자 레벨별 음식 추천 API',
-    description: '사용자 레벨별 음식을 추천합니다.',
-  })
-  @ApiQuery({
-    name: 'userLevel',
-    type: FindUserLevelFoodDto,
-    description: '사용자 레벨',
-  })
-  @ApiCreatedResponse({ description: '레벨별 음식 정보', type: Food })
+  @ApiOperation({ summary: '사용자 레벨별 음식 추천 API' })
   async findUserLevelFoods(
     @Query() param: FindUserLevelFoodDto,
   ): Promise<Food[]> {
-    return this.foodService.findUserLevelFoods(param);
+    const { userLevel } = param;
+    try {
+      const { id: foodLevel } = await this.foodService.findUserLevelFoods(
+        userLevel,
+      );
+      const treeFood = this.foodService.findTreeFood(foodLevel);
+      return treeFood;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Get('/random')
-  @ApiOperation({
-    summary: '음식 랜덤 추천 API',
-    description: '음식을 랜덤 추천합니다.',
-  })
-  @ApiQuery({ name: 'userId', type: String, description: '사용자 ID' })
-  @ApiResponse({ description: '레벨별 음식 정보', type: RandomFoodDto })
+  @ApiOperation({ summary: '음식 랜덤 추천 API' })
   async findRandomFood(
-    @Query() param: { userId: string },
+    @Query() param: FindUserLevelByUserIdDto,
   ): Promise<RandomFoodDto> {
-    return this.foodService.findRandomFoods(param.userId);
+    const { userId } = param;
+    try {
+      const userlevel = await this.foodService.findUserLevel(userId);
+
+      if (userlevel.id === '5') userlevel.id = '4';
+
+      const food = await this.foodService.findFoodByUserLevel(userlevel);
+
+      return food;
+    } catch (e) {
+      throw new HttpException(ERROR_MESSAGE.NOT_FOUND, HttpStatus.BAD_REQUEST);
+    }
   }
 
   @Get(':foodId')
-  @ApiOperation({
-    summary: '음식id 기반으로 해당 음식의 데이터를 가져오는 API',
-    description: '음식 id가 주어진다. 해당 음식의 데이터를 가져온다.',
-  })
-  @ApiParam({ name: '음식 id', type: String })
-  @ApiResponse({
-    description: '음식 리스트',
-    type: PickType(FindFoodDto, [
-      'id',
-      'imageUrl',
-      'logoImageUrl',
-      'name',
-      'subName',
-    ]),
-  })
+  @ApiOperation({ summary: '음식id로 음식 정보를 가져오는 API' })
   async findFoodByFoodId(
-    @Param() param: { foodId: string },
+    @Param() param: FindFoodInfoDto,
   ): Promise<Omit<FindFoodDto, 'hotLevel'>> {
-    return this.foodService.findFoodByFoodId(param.foodId);
+    const { foodId } = param;
+    try {
+      const food = this.foodService
+        .findFoodByFoodId(foodId)
+        .then(({ id, name, subName, imageUrl, logoImageUrl }) => ({
+          id,
+          name,
+          subName,
+          imageUrl,
+          logoImageUrl,
+        }));
+      return food;
```

Comment:
`.then()`을 사용하면 에러 핸들링이 안됩니다.

await으로 받고 처리 부탁드립니다.

```suggestion
      const food = await this.foodService.findFoodByFoodId(foodId)        
      const { id, name, subName, imageUrl, logoImageUrl } = food
     
      return { id, name, subName, imageUrl, logoImageUrl }
```